src/modules/logcollector/src/event_reader_win.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <ctime>
+#include <codecvt>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "reader.hpp"
+#include "winevt_wrapper_win.hpp"
+
+#pragma comment(lib, "wevtapi.lib")
+
+namespace logcollector {
+
+/// @brief Windows Event Tracer Reader class
+class WindowsEventTracerReader : public IReader
+{
+public:
+    /// @brief Constructor for the Windows Event Tracer Reader
+    /// @param logcollector Log collector instance.
+    /// @param channel Channel name.
+    /// @param query Query.
+    /// @param channelRefreshInterval channel query refresh interval in millisecconds.
+    /// @param  .
+    WindowsEventTracerReader(Logcollector &logcollector,
+                            const std::string channel,
+                            const std::string query,
+                            const std::time_t channelRefreshInterval,
+                            std::shared_ptr<IWinAPIWrapper> winAPI = nullptr);
+
+    /// @brief Runs the event reader.
+    /// @return Awaitable result.
+    Awaitable Run() override;
+
+    //@brief Stops the event reader.
+    void Stop() override;
+
+    ///@brief Main function to query events from event channel.
+    Awaitable QueryEvents();
```

Comment:
Shouldn't be private?

Reply:
I'll move it to the private section but it will have impact on the UNit Tests